OrbitGl/LiveFunctionsController.cpp

```diff
@@ -20,6 +22,68 @@ std::pair<uint64_t, uint64_t> ComputeMinMaxTime(
   return std::make_pair(min_time, max_time);
 }
 
+uint64_t AbsDiff(uint64_t a, uint64_t b) {
+  if (a > b) {
+    return a - b;
+  } else {
+    return b - a;
+  }
+}
+
+const TextBox* ClosestTo(TickType point, const TextBox* box_a,
+                         const TextBox* box_b) {
+  uint64_t a_diff = AbsDiff(point, box_a->GetTimer().m_Start);
+  uint64_t b_diff = AbsDiff(point, box_b->GetTimer().m_Start);
+  if (a_diff <= b_diff) {
+    return box_a;
+  }
+  return box_b;
+}
+
+const TextBox* SnapToClosestStart(uint64_t absolute_function_address) {
+  double min_us = GCurrentTimeGraph->GetMinTimeUs();
+  double max_us = GCurrentTimeGraph->GetMaxTimeUs();
+  double center_us = 0.5 * max_us + 0.5 * min_us;
+  TickType center = GCurrentTimeGraph->GetTickFromUs(center_us);
+
+  // First, we find the next function call (text box) that has its end timestamp
+  // after center - 1 (we use center - 1 to make sure that center itself is
+  // included in the timerange that we search). Note that FindNextFunctionCall
+  // uses the end marker of the timer as a timestamp.
+  const TextBox* box = GCurrentTimeGraph->FindNextFunctionCall(
+      absolute_function_address, center - 1);
+
+  // If we cannot find a next function call, then the closest one is the first
+  // call we find before center.
+  if (!box) {
+    return GCurrentTimeGraph->FindPreviousFunctionCall(
+        absolute_function_address, center);
+  }
+
+  // We have to consider the case where center falls into the interior of the
+  // interval of 'box'. In this case, the closest box can be any of two boxes:
+  // 'box' or the next one. It cannot be any box before 'box' because we are
+  // using the start marker to measure the distance.
+  if (box->GetTimer().m_Start <= center && center <= box->GetTimer().m_End) {
```

Comment:
You maybe should erase the comparison: "center <= box->GetTimer().m_End"

First, because you don't need it (99.9999% of times will be true), but second because in the strange case that the middle of the screen is really really near of this m_End point and this center-1 switch its relative position from right to left, you are comparing two text_box whose starting point are at the left of the center instead of the correct ones.

Reply:
Isn't it actually true 100% of the time? My understanding was that FindNextFunctionCall(..., timestamp) will find the next function call with m_End > timestamp. So, here this condition will be true. 

But you are right, even if m_End were center - 1 (which I don't think it can be), then I should still use this case here.

Reply:
Yes, 100%, you are right.

For one moment, I supposed that could be something between center-1 and center at Integer's world.